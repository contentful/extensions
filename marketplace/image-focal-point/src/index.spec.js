import React from 'react';
import { App } from './index';
import { render, cleanup, configure } from '@testing-library/react';

import mockProps from './test/mockProps';

const sdk = {
  ...mockProps.sdk,
  field: {
    getValue: jest.fn(),
    onValueChanged: jest.fn(),
    setValue: jest.fn(),
    removeValue: jest.fn()
  },
  window: {
    startAutoResizer: jest.fn()
  }
};

jest.mock('./utils', () => ({
  getField: jest.fn(),
  isCompatibleImageField: () => true
}));

configure({
  testIdAttribute: 'data-test-id'
});

function renderComponent(sdk) {
  return render(<App sdk={sdk} />);
}

describe('App', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(cleanup);

  it('should read a value from field.getValue() and subscribe for external changes', () => {
    const initialValue = 'x: 0px / y: 0px';
    sdk.field.getValue.mockImplementation(() => initialValue);
    const { getByTestId } = renderComponent(sdk);

    expect(sdk.field.getValue).toHaveBeenCalled();
    expect(sdk.field.onValueChanged).toHaveBeenCalled();
    expect(getByTestId('focal-point').value).toEqual(initialValue);
  });

  it('should call startAutoResizer', () => {
    renderComponent(sdk);
    expect(sdk.window.startAutoResizer).toHaveBeenCalled();
  });

  describe('#render', () => {
    it('should render the extension field view', () => {
      const { container } = renderComponent(sdk);
      expect(container).toMatchSnapshot();
    });
  });
});
