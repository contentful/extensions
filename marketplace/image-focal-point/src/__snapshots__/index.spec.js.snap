// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App #render when there is a valid config should render the extension field view 1`] = `
<div>
  <div
    class="css-1yp4ln"
  >
    <div
      class="TextInput__TextInput___36-K- css-93644d TextInput__TextInput--large___KwY4O TextInput__TextInput--disabled___2t7VS"
    >
      <input
        class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
        data-test-id="focal-point"
        disabled=""
        id="focal-point"
        type="text"
        value="Focal point not set"
      />
    </div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f2k2gl Button__Button--muted___2Wair"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Set focal point
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`App #render when there is an invalid config should render the error message 1`] = `
<div>
  <div
    class="Note__Note___2eSKN Note__Note--negative___3fkwL"
    data-test-id="cf-ui-note"
  >
    <div
      class="Note__Note__icon___20RqC"
    >
      <svg
        class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
        data-test-id="cf-ui-icon"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
    <div>
      <div>
        Could not find a field of type Asset with the ID "
        "
      </div>
    </div>
  </div>
</div>
`;
